client/rest/src/plugins/rosetta/nem/networkRoutes.js

```diff
@@ -0,0 +1,104 @@
+ * along with Catapult.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { getBlockchainDescriptor } from './rosettaUtils.js';
+import { sendJson } from '../../../routes/simpleSend.js';
+import Allow from '../openApi/model/Allow.js';
+import BlockIdentifier from '../openApi/model/BlockIdentifier.js';
+import NetworkIdentifier from '../openApi/model/NetworkIdentifier.js';
+import NetworkListResponse from '../openApi/model/NetworkListResponse.js';
+import NetworkOptionsResponse from '../openApi/model/NetworkOptionsResponse.js';
+import NetworkRequest from '../openApi/model/NetworkRequest.js';
+import NetworkStatusResponse from '../openApi/model/NetworkStatusResponse.js';
+import OperationStatus from '../openApi/model/OperationStatus.js';
+import Peer from '../openApi/model/Peer.js';
+import Version from '../openApi/model/Version.js';
+import { RosettaErrorFactory, rosettaPostRouteWithNetwork } from '../rosettaUtils.js';
+import { NetworkLocator } from 'symbol-sdk';
+import { Network } from 'symbol-sdk/symbol';
+
+export default {
+	register: (server, db, services) => {
+		const blockchainDescriptor = getBlockchainDescriptor(services.config);
+
+		server.post('/network/list', (req, res, next) => {
+			const networkIdentifier = new NetworkIdentifier(blockchainDescriptor.blockchain, blockchainDescriptor.network);
+			const networkListResponse = new NetworkListResponse([networkIdentifier]);
+			return sendJson(res, next)(networkListResponse);
+		});
+
+		server.post('/network/options', rosettaPostRouteWithNetwork(blockchainDescriptor, NetworkRequest, async () => {
+			const getErrorsFromFactoryClass = factory => {
+				const defaultClassPropertyNames = Object.getOwnPropertyNames(class C {});
+				const errorNames = Object.getOwnPropertyNames(factory)
+					.filter(name => !defaultClassPropertyNames.includes(name));
+				return errorNames.map(name => factory[name].apiError);
+			};
+
+			const nodeInfo = await services.proxy.fetch('node/info');
+			const version = new Version();
+			version.rosetta_version = '1.4.13';
+			version.node_version = nodeInfo.metaData.version;
+
+			const isHistoricalNode = features => 2 === (features & 0x2);
+
+			const allow = new Allow();
+			allow.balance_exemptions = [];
+			allow.call_methods = [];
+			allow.errors = getErrorsFromFactoryClass(RosettaErrorFactory);
+			allow.historical_balance_lookup = isHistoricalNode(nodeInfo.metaData.features);
+			allow.mempool_coins = false;
+			allow.operation_statuses = [new OperationStatus('success', true)];
+			allow.operation_types = ['cosign', 'fee', 'multisig', 'transfer'];
+
+			return new NetworkOptionsResponse(version, allow);
+		}));
+
+		server.post('/network/status', rosettaPostRouteWithNetwork(blockchainDescriptor, NetworkRequest, async () => {
+			const [currentBlock, peers, genesisBlock] = await Promise.all([
+				services.proxy.fetch('chain/last-block'),
+				services.proxy.fetch(
+					'node/peer-list/reachable',
```

Comment:
is this intended to be all (active) nodes in the network or only nodes this node is directly communicating with?

Reply:
I was thinking all active, but I can switch to currently communicating with the node.
Which one is returned by Symbol?

Reply:
peers is all active, so this is probably closest to that